### DISCUSSION TITLE
Show history.push (or other methods) in "Actions"

### DISCUSSION THREAD

**Discussion by andreasciamanna** [storybookjs/storybook#18002] [Help]
All the stories are wrapped with a global decorator that handles the routing (something like `<MemoryRouter initialEntries={['/']}>{story()}</MemoryRouter>`).

Probably not needed to specify this, but here is an example of the main App's component:

```tsx
import { createBrowserHistory } from "history";

// ...

<Router history={createBrowserHistory()}>
      <Switch>
// ...

        <Route
          exact
          path="/profile"
          component={Profile}
        />

// ...

      </Switch>
</Router>;
```

Some of these pages (e.g. `Profile`) may need to call `history.push`.
They receive `history: History` as a prop from the `Route` component.

Since the router is different in Storybook, I pass this prop to the args with `createBrowserHistory()`.

So far, so good.

Now, I can't and don't want the story to redirect when  `history.push` is called.

Instead, I'd like to log the call somehow. It would be even better if I could make it show in the "Actions" tab.

So far, I couldn't figure out how to do so.

Any idea of what I could do that?

**Comment by krau5** [comment]
I'm not sure it's actually a good idea to put Router within Storybook.
First of all, it's used to create ur own UI documentation.

**Reply by andreasciamanna** [reply]
Any suggestions on how I could show it at least in "Actions", considering that these pages don't have events props?
